MediaBrowser.MediaEncoding/Encoder/MediaEncoder.cs

```diff
@@ -503,15 +503,7 @@ public Task<string> ExtractVideoImage(string inputFile, string container, MediaS
         {
             var inputArgument = GetInputArgument(inputFile, mediaSource);
 
-            if (isAudio)
-            {
-                if (imageStreamIndex.HasValue && imageStreamIndex.Value > 0)
-                {
-                    // It seems for audio files we need to subtract 1 (for the audio stream??)
```

Comment:
As far as I can tell, all this did was mask the real issue with stream selection, since things would ultimately work fine for any audio files with 1 audio stream and 1 image stream, regardless of which order they're in. (If audio stream is 0 and image stream is 1, we subtract 1, and `-map 0:v0`, or the first video stream. If image stream is 0 and audio stream is 1, we don't subtract 1, and `-map 0:v:0`, and again extract the first video stream).

If there's other context for why this is being done, I'd be curious to know why, but it seems like this is probably a relic from before the Jellyfin fork?

Reply:
It was added 3 years ago, right before Jellyfin forked.
https://github.com/jellyfin/jellyfin/blob/1d7d52ff9e42c3efb4bb2c65e82a4a82faf9decb/MediaBrowser.MediaEncoding/Encoder/MediaEncoder.cs#L636